### DISCUSSION TITLE
Is there a way to cycle between windows of the same application?

### DISCUSSION THREAD

**Discussion by castanonrodrigo** [rxhanson/Charmstone-Community#32] [Q&A]
First of all, thanks for the awesome work on this app! I love using it!
There's just one thing that I think it's missing... is there a way to cycle between opened windows of the same app? Similar to what happens when you click the app icon in the MacOS default dock multiple times.
I know there's a default MacOS shortcut for that( cmd + `) but it doesn't work if one of the windows is in fullscreen or if they are on different desktops.

Thanks!

**Comment by rxhanson** [comment]
Thanks for the feedback. Since Charmstone is sandboxed, I don't think there is a way to do this without offering a non-sandboxed version (outside of the Mac App Store). This is pretty commonly requested, and I do have plans to figure out a path forward on it when I get some time to devote to it.

**Comment by rickarubio** [comment]
Really liking this app -- I think it would be great if there was a way to set shortcuts to an app on a specific monitor, or specific window. In my case, I run multiple windows of chrome on different monitors -- Charmstone currently will only focus back to the most recently selected Chrome window.

Adding more multi-monitor support features would be greatly appreciated, thanks!

**Comment by deconvolved** [comment]
Just adding my interest in this feature should an opportunity arise to address it. I find Charmstone perfect for most of what I need, with the exception of handling multiple windows in an app. My particular pain point is the browser (Chromium derivative). I typically use two displays, and for some reason Charmstone seems to like attaching to a browser window on one display most of the time. Usually the one that I am treating as secondary. So in the typical case, using Charmstone to jump to the browser ends up activating whatever browser window is open on my secondary display. It does this in favor of both the last active window and all windows on the currently selected display.

Thanks for your work on Charmstone. With the exception of my browser woes, I have found it the best option available for my app switching needs.
